browser-interface/packages/shared/comms/adapters/LivekitAdapter.ts

```diff
@@ -40,13 +40,15 @@ export class LivekitAdapter implements MinimumCommunicationsAdapter {
         })
         this.config.logger.log('remote participant left', _.identity)
       })
-      .on(RoomEvent.Disconnected, (_reason: DisconnectReason | undefined) => {
-        this.config.logger.log('disconnected from room', _reason)
-        trackEvent('error', {
-          context: 'livekit-adapter',
-          message: `Got RoomEvent.Disconnected. Reason: ${_reason}`,
-          stack: ""
-        })
+      .on(RoomEvent.Disconnected, (reason: DisconnectReason | undefined) => {
+        this.config.logger.log('disconnected from room', reason)
+        if (reason !== DisconnectReason.CLIENT_INITIATED && reason !== DisconnectReason.DUPLICATE_IDENTITY) {
```

Comment:
@menduz ☝️

Reply:
I think this is very important to be removed:

DisconnectReason.DUPLICATE_IDENTITY

Is triggered by livekit when you re-use an identity for the same user, kicking the previous.

It may be the root cause of our problems. Are we handling disconnects properly? are we trying to re-connect before disconnecting the previous island?

What I propose, is to ONLY filter out the messages of the _disconnected_ rooms. this.disconnected is set upon calling .disconnect()